services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/agent/handler/GetEntityChildrenHandler.java

```diff
@@ -45,10 +48,21 @@ public String handleEvent(ReturnControlEvent event) throws Exception {
 		String synId = ParameterUtils.extractParameter(String.class, "synId", event.getParameters())
 				.orElseThrow(() -> new IllegalArgumentException("Parameter 'synId' of type string is required"));
 
+		String nextPageToken = ParameterUtils.extractParameter(String.class, "nextPageToken", event.getParameters())
+				.orElse(null);
+		String entityType =  ParameterUtils.extractParameter(String.class, "entityType", event.getParameters())
```

Comment:
Can we use the optional branching to handle both cases?  The agent should provide a list of values (the list might only contain one).